Model/Service/Recommendation/StateAwareCategoryService.php

```diff
@@ -247,7 +247,7 @@ function () use ($nostoAccount, $previewMode, $category, $pageNumber, $limit, $f
     /**
      * @inheritDoc
      */
-    public function getLastResult(): ?CategoryMerchandisingResult
+    public function getLastResult(): CategoryMerchandisingResult
```

Comment:
This will result in unhandled exceptions in case `getPersonalisationResult()` fails. `lastResult` will never have a value and other interceptors rely on that. 

Example:
`main.CRITICAL: TypeError: Return value of Nosto\Cmp\Model\Service\Recommendation\StateAwareCategoryService::getLastResult() must be an instance of Nosto\Result\Graphql\Recommendation\CategoryMerchandisingResult, null returned in /magento2/vendor/nosto/module-nostotagging-cmp/Model/Service/Recommendation/StateAwareCategoryService.php:252`

Reply:
You can add a clause to check that the result is instance of `CategoryMerchandisingResult `. But the aftermath will depend on each case.

Reply:
My idea was to initialise empty `CategoryMerchandisingResult` even if there is none. This way null checks are not necessary in interceptors.